packages/react/src/components/AmplifyProvider/index.tsx

```diff
@@ -17,31 +16,72 @@ interface AmplifyProviderProps {
 export function AmplifyProvider({
   children,
   colorMode,
-  theme = defaultTheme,
+  theme,
 }: AmplifyProviderProps) {
   const webTheme = createTheme(theme);
   const { name, cssText } = webTheme;
+
+  // In order for the theme to apply to Portalled elements like our Menu
+  // we need to put the CSS variables we generate from the theme on the
+  // root element. The CSS selector that contains the CSS variables
+  // uses the data attributes present on the root element, and because
+  // the same data attributes are on a div down the DOM tree, the CSS variables
+  // will apply to both.
   React.useEffect(() => {
     if (document && document.documentElement) {
+      // Keep original data attributes to reset on unmount
+      const originalName =
+        document.documentElement.getAttribute('data-amplify-theme');
+      const originalColorMode = document.documentElement.getAttribute(
+        'data-amplify-color-mode'
+      );
       document.documentElement.setAttribute('data-amplify-theme', name);
       document.documentElement.setAttribute(
         'data-amplify-color-mode',
-        colorMode
+        colorMode || ''
       );
+
+      return function cleanup() {
+        document.documentElement.setAttribute(
+          'data-amplify-theme',
+          originalName
+        );
+        document.documentElement.setAttribute(
+          'data-amplify-color-mode',
+          originalColorMode
+        );
+      };
     }
-  }, [theme, colorMode]);
+  }, [name, colorMode]);
```

Comment:
I am worried that the nested theme will still impact the global one. If we have an example using a nested provider there, wouldn't the global theme be affected as well, though we clean that up right after leaving that page

Reply:
The nested theme will override the root data attributes, but because we are also adding them on a div too, the div node will take precedence. The effect would be if you had nested (or just multiple) providers and you referenced the CSS variables outside the DOM tree of the provider (like in the Menu Portal case). I feel like this is a pretty edge case and it is already the current behavior of the provider

Reply:
Just think out loud. I was wondering if adding a `isNested` prop on provider makes sense here and use it to gate the DOM attribute calls so that they don't need to be invoked on nested use cases.

Reply:
Lol yes it does.. and I actually added this in my ThemeProvider PR: https://github.com/aws-amplify/amplify-ui/pull/1446 (need to update the actual prop name). Talking with Scott my thinking is to get this PR merged in to fix our issues we are currently having and then spend some more time on the ThemeProvider PR. The changes this PR has are similar to the ThemeProvider PR too, but with just no API changes.